File: src/Flip to Maximize 1s/Code/solution.py
class Solution:
    def maxOnes(self, arr):
        ones = 0
        
        # Count current number of 1s
        for num in arr:
            if num == 1:
                ones += 1
        
        current_gain = 0
        max_gain = 0
        
        # Apply Kadane's Algorithm
        for num in arr:
            # 0 gives +1 gain, 1 gives -1 loss
            value = 1 if num == 0 else -1
            
            current_gain = max(value, current_gain + value)
            max_gain = max(max_gain, current_gain)
        
        return ones + max_gain

File: src/Flip to Maximize 1s/Code/solution.cpp
class Solution
{
public:
    int maxOnes(vector<int> &arr)
    {
        int ones = 0;

        // Count existing 1s
        for (int num : arr)
        {
            if (num == 1)
            {
                ones++;
            }
        }

        int currentGain = 0;
        int maxGain = 0;

        // Kadane's Algorithm
        for (int num : arr)
        {
            int value;

            // 0 gives +1 gain, 1 gives -1 loss
            if (num == 0)
            {
                value = 1;
            }
            else
            {
                value = -1;
            }

            currentGain = max(value, currentGain + value);
            maxGain = max(maxGain, currentGain);
        }

        return ones + maxGain;
    }
};